### PR TITLE
DCMTK: init at 3.6.4

### PR DESCRIPTION
**Pull request by iimog**
###### Motivation for this change
Add the DICOM Toolkit DCMTK as package. It is a command line suite to handle medical imaging data in DICOM format. Including inspection, modification, and conversion.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, zlib, libtiff, libxml2, openssl, libiconv, libpng, cmake }:
+
+with stdenv.lib;
+stdenv.mkDerivation {
+  pname = "DCMTK";
```
Comment (prusnak) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: This should probably be "dcmtk"?

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = {
+    description = "DICOM Toolkit";
+    longDescription = ''
+      DCMTK is a collection of libraries and applications implementing large parts of the DICOM standard.
+      It includes software for examining, constructing and converting DICOM image files, handling offline media, sending and receiving images over a network connection, as well as demonstrative image storage and worklist servers.
+      DCMTK is is written in a mixture of ANSI C and C++.
+      It comes in complete source code and is made available as "open source" software.
+    '';
+    homepage = https://dicom.offis.de/dcmtk.php.en;
```
Comment (prusnak) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: homepage = "https://dicom.offis.de/";

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, zlib, libtiff, libxml2, openssl, libiconv, libpng, cmake }:
+
+with stdenv.lib;
+stdenv.mkDerivation {
+  pname = "DCMTK";
+  version = "3.6.4";
+  src = fetchFromGitHub {
+      owner = "DCMTK";
+      repo = "dcmtk";
+      rev = "1967b13134308f311e6a827e616958c6a4da5bc9";
```
Comment (prusnak) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: `rev = "DCMTK-${version}";`

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, zlib, libtiff, libxml2, openssl, libiconv, libpng, cmake }:
+
+with stdenv.lib;
+stdenv.mkDerivation {
+  pname = "DCMTK";
+  version = "3.6.4";
+  src = fetchFromGitHub {
+      owner = "DCMTK";
+      repo = "dcmtk";
```
Comment (prusnak) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: `repo = pname;`

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      rev = "1967b13134308f311e6a827e616958c6a4da5bc9";
+      sha256 = "0fbx35zax8n4gayaac5bankqwzg2y2adggykbbf8lpd773jfxsp6";
+  };
+
+  buildInputs = [ cmake libpng zlib libtiff libxml2 openssl libiconv ];
```
Comment (jonringer) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: cmake should be in nativeBuildInputs
```suggestion
  nativeBuildInputs = [ cmake ];
  buildInputs = [ cmake libpng zlib libtiff libxml2 openssl libiconv ];
```

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = {
+    description = "DICOM Toolkit";
```
Comment (jonringer) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: not really descriptive, maybe take from the long description.
```suggestion
    description = "Collection of libraries and applications implementing large parts of the DICOM standard";
```

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    longDescription = ''
+      DCMTK is a collection of libraries and applications implementing large parts of the DICOM standard.
+      It includes software for examining, constructing and converting DICOM image files, handling offline media, sending and receiving images over a network connection, as well as demonstrative image storage and worklist servers.
```
Comment (jonringer) [inline pkgs/applications/science/medicine/dcmtk/default.nix]: this line is too long
```suggestion
      It includes software for examining, constructing and converting DICOM image files, handling offline media, 
      sending and receiving images over a network connection, as well as demonstrative image storage and worklist servers.
```

**Comment by iimog** [comment]
Thanks @prusnak and @jonringer for your reviews. I agree with all your suggestions. What is the preferred way to incorporate them. Should I make these changes and force push them so it remains a single commit? In this case you will probably not be credited for you suggested changes (or is there a way to do that?).

**Comment by prusnak** [comment]
Yes, amend the commit and force push to branch

**Comment by prusnak** [comment]
LGTM

**Review by rasendubi — COMMENTED** [review]
Minor nitpicks. Otherwise, looks good.

Builds fine on NixOS with sandboxing.

**Comment by iimog** [comment]
@rasendubi thanks for catching that. I just force pushed your suggested changes

File: pkgs/applications/science/medicine/dcmtk/default.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  meta = {
+    description = "Collection of libraries and applications implementing large parts of the DICOM standard";
+    longDescription = ''
+      DCMTK is a collection of libraries and applications implementing large parts of the DICOM standard.
+      It includes software for examining, constructing and converting DICOM image files, handling offline media,
+      sending and receiving images over a network connection, as well as demonstrative image storage and worklist servers.
+      DCMTK is is written in a mixture of ANSI C and C++.
+      It comes in complete source code and is made available as "open source" software.
+    '';
+    homepage = "https://dicom.offis.de/";
```
[the remark this answers is not in the document]
Comment (prusnak) [reply pkgs/applications/science/medicine/dcmtk/default.nix]: Ah, sorry, that was my mistake. The new suggestion is nice.

**Comment by rasendubi** [comment]
@GrahamcOfBorg build dcmtk

**Pull request merged**